src/sssom/util.py

```diff
@@ -1397,18 +1417,26 @@ def invert_mappings(
         non_predicate_modified_df = df
 
     if subject_prefix:
-        subject_starts_with_prefix_condition = df[SUBJECT_ID].str.startswith(subject_prefix + ":")
-        object_starts_with_prefix_condition = df[OBJECT_ID].str.startswith(subject_prefix + ":")
+        # Filter rows where 'SUBJECT_ID' starts with the prefix but 'OBJECT_ID' does not
         prefixed_subjects_df = pd.DataFrame(
             non_predicate_modified_df[
-                (subject_starts_with_prefix_condition & ~object_starts_with_prefix_condition)
+                (
+                    non_predicate_modified_df[SUBJECT_ID].str.startswith(subject_prefix + ":")
+                    & ~non_predicate_modified_df[OBJECT_ID].str.startswith(subject_prefix + ":")
+                )
             ]
         )
+
+        # Filter rows where 'SUBJECT_ID' does not start with the prefix but 'OBJECT_ID' does
         non_prefix_subjects_df = pd.DataFrame(
             non_predicate_modified_df[
-                (~subject_starts_with_prefix_condition & object_starts_with_prefix_condition)
+                (
+                    ~non_predicate_modified_df[SUBJECT_ID].str.startswith(subject_prefix + ":")
+                    & non_predicate_modified_df[OBJECT_ID].str.startswith(subject_prefix + ":")
+                )
```

Comment:
Why these changes above? Are they related to warning?

Reply:
Yes: `UserWarning: Boolean Series key will be reindexed to match DataFrame index.`